src/API/NavigationFavorites.php

```diff
@@ -0,0 +1,206 @@
+<?php
+/**
+ * REST API Navigation Favorites controller
+ *
+ * Handles requests to the navigation favorites endpoint
+ */
+
+namespace Automattic\WooCommerce\Admin\API;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Admin\Features\Navigation\Favorites;
+
+/**
+ * REST API Favorites controller class.
+ *
+ * @extends WC_REST_CRUD_Controller
+ */
+class NavigationFavorites extends \WC_REST_Data_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-admin';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'navigation/favorites';
+
+	/**
+	 * Register the routes
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
```

Comment:
I think these are good endpoints, but defaulting to the current user I think breaks convention a bit here.

I know it's a few extra lines of code now, but I think it's worthwhile to add 3 more endpoints with `/me` suffixes to all of these endpoints that defaults to the current user.

Reply:
Thanks Josh, good call. I've added the `/me` endpoints and changed these to require the `user_id` parameter. 

One curiosity is that as it is now the `/me` endpoints will still technically accept and utilize the `user_id` parameter since they're using the same handler function. They're just not required; is that considered a bad thing? I can break those out, although it seems elegant to be able to utilize the same handler if it's not a concern.

Reply:
It actually occurred to me that this could be a security flaw as is, addressing now.

Reply:
I would probably break them out to separate functions, but we can reuse the primary function.  For example:

```
get_items( $request ) {
  // original logic to get items by given user id
}

get_items_by_current_user( $request ) {
  $user_id = get_current_user_id();
  // If user not logged in, return error
  $request->set_param( 'user_id', $user_id );
  get_items( $request );
}

Reply:
I agree, that's a bit clearer. Done.